be/src/olap/reader.cpp

```diff
@@ -613,15 +613,14 @@ Status TabletReader::_init_delete_condition(const ReaderParams& read_params) {
     }
 
     auto delete_init = [&]() -> Status {
-        return _delete_handler.init(_tablet_schema, _tablet->delete_predicates(),
+        return _delete_handler.init(_tablet_schema, read_params.delete_predicates,
                                     read_params.version.second, this);
     };
 
     if (read_params.reader_type == READER_ALTER_TABLE) {
         return delete_init();
     }
 
```

Comment:
If not need acquire tablet lock here, then I think we do not need distinguish alter table and others. 620-623 could be deleted.

Reply:
fixed